doc/cabal-package.rst

```diff
@@ -1017,24 +1017,70 @@ disambiguation purposes. Example:
     $ cabal repl bench:baz
 
 Freezing dependency versions
-""""""""""""""""""""""""""""
-
-If a package is built in several different environments, such as a
-development environment, a staging environment and a production
-environment, it may be necessary or desirable to ensure that the same
-dependency versions are selected in each environment. This can be done
-with the ``freeze`` command:
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
 .. code-block:: console
 
     $ cabal freeze
 
-The command writes the selected version for all dependencies to the
-``cabal.config`` file. All environments which share this file will use
-the dependency versions specified in it.
+generates ``cabal.project.freeze`` file, which describes the exact dependency tree as it was resolved at that moment by Cabal.
+This means it captures an exact version of every dependency, including dependencies of dependencies, recursively all the way.
```

Comment:
I think you definitely want to include the caveats from https://github.com/haskell/cabal/issues/8059 that this is a _platform-specific_ freeze, and as such is not necessarily usable by other people.

Reply:
I think I mostly agree with this comment: https://github.com/haskell/cabal/issues/8059#issuecomment-1076892558